tests/test_coerce.py

```diff
@@ -0,0 +1,39 @@
+import sys
+
+import pytest
+
+from ring.func.base import coerce
+
+try:
+    from dataclasses import dataclass  # noqa
+except ImportError:  # pragma: no cover
+    pass
+
+
+class RingKey(object):
```

Comment:
`RingKey` sounds like it is an implmentation of ring key interface.
Something like `UserObject` will be better name.